### PR TITLE
refactor: migrate core22 store commands

### PR DESCRIPTION
**Pull request by mr-cal**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `tox run -m lint`?
- [x] Have you successfully run `tox run -e test-py310`? (supported versions: `py39`, `py310`, `py311`, `py312`)

-----

Migrate core22 commands related to the store from `craft_cli.BaseCommand` to `craft_application.commands.AppCommand`.

This allows store operations for core24 snaps.  It also fixes a behavior where commands unrelated to a project would fail.  For example, running `snapcraft login` in a core24 project directory would fail with `'login' is not implemented in core24!`

Pycharm's refactor utility did not help but it was straightforward enough to do manually.

### CODE REVIEW

**Comment by mr-cal** [comment]
> Will craft-application replace those commands later?

I'm not sure, since craft-store hasn't been integrated into craft-application.

File: snapcraft/application.py [hunk 1]
```diff
@@ -224,21 +224,21 @@ def create_app() -> Snapcraft:
...
-            unimplemented.Release,
-            unimplemented.Close,
-            unimplemented.Status,
-            unimplemented.Upload,
+            commands.StoreReleaseCommand,
+            commands.StoreCloseCommand,
+            commands.StoreStatusCommand,
+            commands.StoreUploadCommand,
             unimplemented.Push,
             unimplemented.Promote,
```
Comment (tigarmo) [inline snapcraft/application.py:232]: why not promote?

Comment (mr-cal) [reply snapcraft/application.py:232]: There are 11 legacy store commands that I was going to do in a separate PR.  Looking at them now, it looks straightforward enough. I'll try to fit it into this PR.

Comment (mr-cal) [reply snapcraft/application.py]: Added!

File: tools/docs/gen_cli_docs.py [hunk 2]
```diff
@@ -99,7 +99,7 @@ def main(docs_dir):
         g = group_path.open("w")
 
         for cmd_class in sorted(group.commands, key=lambda c: c.name):
-            cmd = cmd_class({})
+            cmd = cmd_class({"app": {}, "services": {}})
```
Comment (sergiusens) [inline tools/docs/gen_cli_docs.py:102]: maybe add a comment on why this is needed

**Review by sergiusens — APPROVED** [review]
:+1:

**Pull request merged**